xla/backends/gpu/runtime/dynamic_slice_thunk.h

```diff
@@ -57,8 +58,10 @@ class DynamicSliceThunk : public Thunk {
       std::vector<std::optional<std::vector<Offset>>> offsets,
       std::vector<std::optional<Shape>> orig_shapes,
       std::vector<std::optional<Shape>> sliced_shapes,
-      std::vector<std::optional<uint64_t>> offset_byte_sizes);
-
+      std::vector<std::optional<uint64_t>> offset_byte_sizes,
+      std::vector<std::unique_ptr<HloModule>> extracted_offset_modules = {},
```

Comment:
Should we have a `struct` wrapping these three vectors?

After all, they only exist together, or not at all.